src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -172,7 +172,11 @@ PaymentSandbox::balanceHook (AccountID const& account,
         }
```

Comment:
Not your fault, but the documentation of what `balanceHook()` is supposed to do is pretty sparse.  I hunted up 3 levels of inheritance and finally found this;
```
    // Called to adjust returned balances
    // This is required to support PaymentSandbox
```
https://github.com/ripple/rippled/blob/develop/src/ripple/ledger/ReadView.h#L306-L307.  While I can see that documentation is appropriate at the `ReadView.h` level, here in PaymentSandbox there should be more text describing the intent of the `balanceHook()` adjustments.

There is a great paragraph starting on line 136.  But that paragraph describes the _differences_ between two algorithms.  I don't get an overview of what both (similar) algorithms are intended to accomplish.  I guess I'm looking for a sentence that says, "balanceHook() returns ..." and then describes how this balance is different from just getting the balance from the sle.